helpers/-for-of.js

```diff
@@ -14,6 +14,23 @@ var bindAndRead = function (value) {
 	}
 };
 
+function forOfInteger(integer, variableName, options) {
+	var result = [];
+  for (var i = 0; i < integer; i++) {
+		var variableScope = {};
+		if(variableName !== undefined){
+			variableScope[variableName] = i;
+		}
+		result.push(
+			options.fn( options.scope
+				.add({ index: i }, { special: true })
+				.addLetContext(variableScope) )
```

Comment:
Do you think we should create a variable?

```
{{# for(ittr of this.integerValue) }}
  <li>{{scope.index}}: {{ ittr }}</li>
{{/ for }}
```

I sorta think access should be only through `scope.index` like: `{{# for( this.integerValue ) }} {{scope.index}} {{/for}}`.

Reply:
On second thought, I think it's ok to make `{{# for(ittr of this.integerValue) }}` work.  But I wouldn't show this as the main way of using it.